src/features/devices/soil_humidity/soil_moisture_sensor.py

```diff
@@ -0,0 +1,57 @@
+from features.devices.device import Device, DeviceConfig, Topic
+from features.logger.logger import Logger
+from features.mqtt.mqtt import BaseMqttClient
+from utime import ticks_ms, ticks_diff, time
+from ujson import dumps, loads
+from features.analog_accessor.analog_accessor import AnalogAccessor
+import machine
+from drivers.sms import SMSDriver
+import uasyncio
+
+
+class Moisture(Topic):
+    pass
+
+
+class SoilMoistureSensor(Device):
+    SMS_READ_SPAN = 2000
+
+    def __init__(self, mqtt: BaseMqttClient,
+                 config: DeviceConfig,
+                 logger: Logger,
+                 analog_accessor: AnalogAccessor):
+        super().__init__(mqtt, config, logger)
+        data = config.config
+
+        self._moisture = Moisture(mqtt, data["moisture"])
+        self.loop_span_ms = data["loopSpanMs"]
+        channel = data["channel"]
+        self._sensor = SMSDriver(analog_accessor, channel)
+
+        self._last_read = ticks_ms() - self.SMS_READ_SPAN
+
+
+    async def _update_config(self):
+        pass
+
+    async def _loop(self):
+        current_time = ticks_ms()
+
+        if abs(ticks_diff(current_time, self._last_read)) < SoilMoistureSensor.SMS_READ_SPAN:
+            await uasyncio.sleep_ms(self.loop_span_ms)
```

Comment:
Ten await jest nie potrzebny tutaj dodakowo, dla DHT11 ma to znaczenie bo on w swoich wlasciwosciach ma ze mozna zrobic odczyt raz na 2-3 sekundy. - a tutaj chyba to bezproblemowo działa